R/preview.R

```diff
@@ -248,15 +256,23 @@ st2article <- function(..., .list = NULL, ntex = 1,  #nocov start
   temp <- mgluet(temp, .envir = env)
 
   if(is.character(caption)) {
-    wrap_with_caption <- function(text, i) {
-      if(is.null(i)) i <- "<no name given>"
-      short <- paste0("pmtables output preview - ", i)
+    wrap_with_caption <- function(text, i, short, cap) {
+      if(is.character(cap_main(text))) {
+        caption <- cap_main(text)
+        short <- cap_short(text)
+        if(is.null(short)) short <- caption
+      }
       pt_wrap(
-        text, context = "tex", caption = caption, short = short,
-        con = NULL
+        text, context = "tex", caption = caption, short = short, con = NULL
       )
     }
-    tables <- imap(tables, wrap_with_caption)
+    tables <- Map(
```

Comment:
Need to switch this from `imap()` since we're mapping along vectors of shorts and captions.